src/pages/loading-waiting-states/spinner-overlays/index.tsx

```diff
@@ -0,0 +1,197 @@
+import * as Colors from '@pxblue/colors';
+
+const useStyles = makeStyles((theme: Theme) => ({
+    container: {
+        padding: `${theme.spacing(2)}px ${theme.spacing(2)}px`,
+        maxWidth: '600px',
+        margin: '0 auto',
+        height: `calc(100vh - ${theme.spacing(8)}px)`,
+        [theme.breakpoints.down('xs')]: {
+            height: `calc(100vh - ${theme.spacing(7)}px)`,
+        },
+    },
+    toolbarGutters: {
+        padding: `0 ${theme.spacing(2)}px`,
+    },
+    eulaContainerStyle: {
+        height: `calc(100vh - 100px)`,
+        position: 'relative',
+        padding: 0,
+    },
+    loadingOverlayContainerStyle: {
+        display: 'flex',
+        position: 'absolute',
+        top: 0,
+        left: 0,
+        width: '100%',
+        height: '100%',
+        zIndex: 4,
+        justifyContent: 'center',
+        alignItems: 'center',
+        flexDirection: 'column',
+    },
+    eulaDetailsStyle: {
+        flex: '1 1 0',
+        display: 'flex',
+        flexDirection: 'column',
+        height: '100%',
+    },
+    eulaDetailSectionStyle: {
+        overflow: 'auto',
+        display: 'flex',
+        flexDirection: 'column',
+        flex: '1 1 0',
+        padding: `${theme.spacing(3)}px`,
+    },
+    loadingContainerStyle: {
+        height: '100%',
+        paddingTop: `${theme.spacing(1)}px`,
+    },
+    overlayContainerStyle: {
+        position: 'absolute',
+        top: 0,
+        left: 0,
+        width: '100%',
+        height: '100%',
+        background: Colors.white[200],
+        zIndex: 3,
+        opacity: 0.2,
+    },
+    btnContainerStyle: {
+        padding: `${theme.spacing(3)}px`,
+    },
+    reloadButtonStyle: {
+        width: '100%',
+    },
+    eulaContentStyle: {
+        paddingTop: `${theme.spacing(1)}px`,
+    },
+    eulaConfirmationCheckStyle: {
+        padding: `0 ${theme.spacing(3)}px`,
+    },
+}));
+
+export const SpinnerOverlays = (): JSX.Element => {
+    const dispatch = useDispatch();
+    const theme = useTheme();
+    const classes = useStyles();
+    const spinnerCirlceSize = 96;
+    const [checkboxState, setCheckboxState] = useState(false);
+    const [eulaLoaded, setEulaLoaded] = useState(false);
+
+    const changeCheckboxState = (event: React.ChangeEvent<HTMLInputElement>): void => {
+        setCheckboxState(event.target.checked);
+    };
+
+    useEffect(() => {
+        let isEulaLoaded = true;
+
+        setTimeout(() => {
+            if (isEulaLoaded) {
+                setEulaLoaded(true);
+            }
+        }, 3000);
+
+        return (): void => {
+            isEulaLoaded = false;
+        };
+    }, [checkboxState, eulaLoaded, setEulaLoaded]);
+
+    const reloadEulaDetails = useCallback((): void => {
+        setEulaLoaded(false);
+        setCheckboxState(false);
+    }, [checkboxState, eulaLoaded, setEulaLoaded]);
+
+    return (
+        <div style={{ backgroundColor: theme.palette.background.paper, minHeight: '100vh' }}>
```

Comment:
I think you can leave the background color off here. White on white makes it so the card doesn't pop off of the page like it's supposed to. We can stick with the default off-white background for the page that comes from the themes.